modules/cockroachdb/cockroachdb_test.go

```diff
@@ -144,6 +147,71 @@ func (suite *AuthNSuite) TestQuery() {
 	suite.Equal(523123, id)
 }
 
+// TestWithWaitStrategyAndDeadline covers a previous regression, container creation needs to fail to cover that path.
+func (suite *AuthNSuite) TestWithWaitStrategyAndDeadline() {
+
+	contextDeadlineExceeded := fmt.Errorf("failed to start container: context deadline exceeded")
+	nodeStartUpCompleted := "node startup completed"
+	suite.Run("Expected Failure To Run", func() {
+
+		ctx := context.Background()
+
+		// This will never match a log statement
+		suite.opts = append(suite.opts, testcontainers.WithWaitStrategyAndDeadline(time.Millisecond*250, wait.ForLog("Won't Exist In Logs")))
+		container, err := cockroachdb.RunContainer(ctx, suite.opts...)
+		suite.Require().Error(err, contextDeadlineExceeded)
+		suite.T().Cleanup(func() {
+			if container != nil {
+				err := container.Terminate(ctx)
+				suite.Require().NoError(err)
+			}
+		})
+
+	})
+
+	suite.Run("Expected Failure To Run But Would Succeed ", func() {
+
+		ctx := context.Background()
+
+		// This will timeout as we didn't give enough time for intialization, but would have succeeded otherwise
+		suite.opts = append(suite.opts, testcontainers.WithWaitStrategyAndDeadline(time.Millisecond*20, wait.ForLog(nodeStartUpCompleted)))
+		container, err := cockroachdb.RunContainer(ctx, suite.opts...)
+		suite.Require().Error(err, contextDeadlineExceeded)
+		suite.T().Cleanup(func() {
+			if container != nil {
+				err := container.Terminate(ctx)
+				suite.Require().NoError(err)
+			}
+		})
+
+	})
+
+	suite.Run("Succeeds And Executes Commands", func() {
+
+		ctx := context.Background()
+
+		// This will succeed
+		suite.opts = append(suite.opts, testcontainers.WithWaitStrategyAndDeadline(time.Second*60, wait.ForLog(nodeStartUpCompleted)))
+		container, err := cockroachdb.RunContainer(ctx, suite.opts...)
+		suite.Require().NoError(err)
+
+		conn, err := conn(ctx, container)
+		suite.Require().NoError(err)
+		defer conn.Close(ctx)
```

Comment:
Why no error checking to see if db connection was closed successfully?

Reply:
Defer is going to run at the end of scope, you aren't able to do this.

Reply:
That's true, but you could defer (and run) an anonymous function, and in that anonymous function, you could close the database and then error check to see if it closed right.

That said, I'm not asking this change be made, especially if the rest of the tests aren't doing it like this. Was just curious because I'm new-ish to Golang so I am trying to understand best practices.

Feel free to resolve this thread.

Reply:
Indeed, t.Cleanup could be of interest. Thanks @AhmedNSidd fro bringing this point to the table.